modules/webhooks/spec/components/webhooks/outgoing/webhooks/row_component_spec.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "rails_helper"
+
+RSpec.describe Webhooks::Outgoing::Webhooks::RowComponent, type: :component do
+  def render_component(...)
+    render_inline(described_class.new(...))
+  end
+
+  let(:webhook) { build_stubbed(:webhook, name: "Hook me up", all_projects:) }
+  let(:table) do
+    instance_double(Webhooks::Outgoing::Webhooks::TableComponent, columns: [column],
+                                                                  target_controller: "webhooks/outgoing/admin")
+  end
+  let(:all_projects) { true }
+
+  subject(:rendered_component) do
+    render_component(row: webhook, table:)
+  end
```

Comment:
Something like this, with less indirections:

```suggestion
  subject(:rendered_component) do
    table = instance_double(Webhooks::Outgoing::Webhooks::TableComponent,
                            columns: [column],
                            target_controller: "webhooks/outgoing/admin")
    render_inline(described_class.new(row: webhook, table:))
  end

```